keps/sig-network/20200311-add-label.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Add property filed in PodIP of PodStatus
+authors:
+  - "@s1061123"
+  - "@dosmith"
+owning-sig: sig-network
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# Service for custom controllers
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories [optional]](#user-stories-optional)
+    - [Endpoint Selection for the certain service (IPv4/IPv6)](#endpoint-selection-for-the-certain-service-ipv4ipv6)
+    - [Multiple Service/Endpoint Controller in the cluster](#multiple-serviceendpoint-controller-in-the-cluster)
+- [Design Details](#design-details)
+  - [PodStatus API Change](#podstatus-api-change)
+  - [Provide well-known property](#provide-well-known-property)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+This KEP is to add property field in PodIP structure, to add additional information for Pod IP
+and to provide mechanism to flexible endpoint/endpointslice manages among multiple controllers.
+
+## Motivation
+
+Currently Kubernetes Pod has multiple IP addresses, based on [sig-network/20180612-ipv4-ipv6-dual-stack.md](https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20180612-ipv4-ipv6-dual-stack.md). In the kep, `property` field is also discussed to add additional information, such as interface name, gateway etc, even though it is not implemented yet. 
+
+### Goals
+
+   * Provide flexible way for custom endpoint/endpointslice controller to pick up IPs from Pod, based on properties
+   * Provide the properties, `service.kubernetes.io/service-controller-name` to specify which the endpoint/endpointslice controller use the IP for endpoint/endpointslice
+
+### Non-Goals
+
+   * This KEP does not change any semantics in the Endpoint/EndpointSlice 
+
+## Proposal
+
+This KEP is to add property field in PodIP of PodStatus to add information for Pod IP
+and to provide mechanism to flexible endpoint/endpointslice manages among multiple controllers.
+
+### User Stories [optional]
+
+#### Endpoint Selection for the certain service (IPv4/IPv6)
+
+In IPv4/v6 dual-stack case, even though pod has multiple IP, user may want to provide service IPv6 only
```

Comment:
The dual-stack work includes a specific provision for this (`ipFamily`) which negates this use-case.